core/src/main/java/com/radixdlt/monitoring/InMemorySystemInfo.java

```diff
@@ -66,48 +66,22 @@
 
 import com.google.inject.Inject;
 import com.radixdlt.consensus.LedgerProof;
-import com.radixdlt.consensus.QuorumCertificate;
-import com.radixdlt.consensus.bft.*;
-import com.radixdlt.consensus.epoch.EpochChange;
+import com.radixdlt.consensus.bft.Round;
 import com.radixdlt.consensus.epoch.EpochRound;
-import com.radixdlt.consensus.liveness.EpochLocalTimeoutOccurrence;
-import com.radixdlt.constraintmachine.REEvent.ValidatorBFTDataEvent;
 import com.radixdlt.environment.EventProcessor;
 import com.radixdlt.ledger.LedgerUpdate;
 import com.radixdlt.rev2.LastEpochProof;
-import com.radixdlt.rev2.LastProof;
-import java.util.Optional;
 import java.util.concurrent.atomic.AtomicReference;
 
 /** Manages system information to be consumed by clients such as the api. */
 public final class InMemorySystemInfo {
-  private final AtomicReference<EpochLocalTimeoutOccurrence> lastTimeout = new AtomicReference<>();
   private final AtomicReference<EpochRound> currentEpochRound =
       new AtomicReference<>(EpochRound.of(0L, Round.genesis()));
-  private final AtomicReference<QuorumCertificate> highQC = new AtomicReference<>();
-  /* BAB-TODO: Add back in missed proposal tracking
-  private final AtomicMarkableReference<Optional<ValidatorBFTDataEvent>> missedProposals =
-      new AtomicMarkableReference<>(Optional.empty(), false);
-   */
-  private final AtomicReference<LedgerProof> ledgerProof;
   private final AtomicReference<LedgerProof> epochsLedgerProof;
-  private final SelfValidatorInfo self;
-  // private final RadixEngine<LedgerAndBFTProof> radixEngine;
 
   @Inject
-  public InMemorySystemInfo(
-      @LastProof LedgerProof lastProof,
-      @LastEpochProof LedgerProof lastEpochProof,
-      SelfValidatorInfo self
-      /*RadixEngine<LedgerAndBFTProof> radixEngine*/ ) {
-    this.ledgerProof = new AtomicReference<>(lastProof);
+  public InMemorySystemInfo(@LastEpochProof LedgerProof lastEpochProof) {
```

Comment:
_(note from author)_

As a drive-by refactor, I have trimmed down this `SystemInfo` thing a bit.
Is anyone else under the impression that it should really vanish completely?

Reply:
A lot of the stuff here is related to the old Olympia version of the "own missedProposals" metric which I think you've handled elsewhere?

I don't have a strong feeling about whether this should still exist or not. Feels like an okay abstraction to keep, or move into something else.